src/data/markdown/translated-guides/en/07 Misc/05 k6 Extensions.md

```diff
@@ -1,4 +1,197 @@
+and [xk6-influxdbv2](https://github.com/li-zhixin/xk6-influxdbv2).
+
+
+## How to get started
+
+There are two ways that xk6 can be used:
+
+- By k6 users that wish to enhance their tests with existing extensions. A
+  familiarity with the command line and Go is preferred, but not required.
+
+- By Go developers interested in creating their own k6 extension. They'll need to be
+  familiar with both Go and JavaScript, understand how the k6 Go<->JS bridge works,
+  and maintain a public repository for the extension that keeps up to date with any
+  breaking API changes while xk6 is being stabilized.
+
+
+### Using xk6 to build a k6 binary
+
+You might have found a neat k6 extension on the [Ecosystem page](/ecosystem) or on
+[GitHub](https://github.com/topics/xk6) and wish to use it in your tests. Great! The
+process is relatively simple and we're working on simplifying it further for end
+users.
+
+You'll first need to setup [Go](https://golang.org/doc/install) and
+[Git](https://git-scm.com/). Make sure that your `$PATH` environment variable is
+updated and that `go version` returns the correct version.
+
+Then install xk6 with:
+
+<CodeGroup labels={[]} lineNumbers={[false]}>
+
+```bash
+$ go install github.com/k6io/xk6/cmd/xk6@latest
+```
+
+</CodeGroup>
+
+And confirm that `which xk6` on Linux/macOS or `where xk6.exe` on Windows returns a
+valid path. Otherwise ensure that `$GOPATH` is correctly defined and that
+`$GOPATH/bin` is added to your `$PATH` environment variable. See the
+[Go documentation](https://golang.org/cmd/go/#hdr-GOPATH_environment_variable) for details.
+
+Once xk6 is installed, building a k6 binary with one or more extensions can be done
+with the following command:
+
+<CodeGroup labels={[]} lineNumbers={[false]}>
+
+```bash
+$ xk6 build v0.33.0 \
+  --with github.com/imiric/xk6-sql \
+  --with github.com/szkiba/xk6-prometheus
+```
+
+</CodeGroup>
+
+This will build a `k6` binary in the current directory based on k6 v0.33.0, bundling
+a JavaScript and an Output extension. Now you can run a script with this binary that
+uses the [`xk6-sql` JS API](https://github.com/imiric/xk6-sql) and the
+[Prometheus output](https://github.com/szkiba/xk6-prometheus).
+
+Note that you'll need to specify the binary in the current directory (i.e. `./k6` or
+`.\k6` on Windows) to avoid using any other `k6` binary in your `PATH`.
+
+
+### Writing a new extension
+
+A good starting point for using xk6 and writing your own extension is the [xk6
+introductory article](https://k6.io/blog/extending-k6-with-xk6).
+<!-- TODO: How much of this article do we want to reproduce here? -->
+
+You'll first need to decide the type of extension you wish to create. Choose a
+JavaScript extension if you want to extend the JS functionality of your script, or an
+Output extension if you want to process the metrics emitted by k6 in some way.
+
+A few things to keep in mind when writing a new extension:
+<!-- TODO: Better structure this list? -->
```

Comment:
Agree. It is a bit hard to follow. 

I wonder if. it could be good to show snippets of relevant extension code to distinguish between the various sections and make the whole section more pleasant to read.

Reply:
Yeah, could be... In #169 we kind of agreed to not show any Go code here, but we could split this section into "Writing a new JS extension" (linking to the existing article) and "Writing a new Output extension", and do a small walkthrough with some snippets of code instead of a complete extension like in the article.

I'll think about it, and more opinions are welcome. :)

Reply:
@ppcano Take a look at the latest changes. It kind of breaks the no-code rule, but I really don't think we can document writing an extension without showing code... :)